File: app/templates/app-java/src/main/java/di/components/MainComponent.java
package <%= appPackage %>.di.components;

import dagger.Component;
import <%= appPackage %>.di.ActivityScope;
import <%= appPackage %>.di.modules.MainModule;
import <%= appPackage %>.ui.main.MainActivity;
import <%= appPackage %>.ui.main.MainFragment;

@ActivityScope
@Component(dependencies = {ApplicationComponent.class}, modules = {MainModule.class})
public interface MainComponent {

    void inject(MainActivity activity);
    void inject(MainFragment fragment);

}

File: app/templates/app-kotlin/src/main/java/di/components/MainComponent.kt
package <%= appPackage %>.di.components

import dagger.Component
import <%= appPackage %>.di.ActivityScope
import <%= appPackage %>.di.modules.MainModule
import <%= appPackage %>.ui.main.MainActivity
import <%= appPackage %>.ui.main.MainFragment

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(MainModule::class))
interface MainComponent {

  fun inject(activity: MainActivity)
  fun inject(fragment: MainFragment)

}

